dashboard/src/actions/repos.ts

```diff
@@ -116,16 +116,10 @@ export const deleteRepo = (
 ): ThunkAction<Promise<boolean>, IStoreState, null, AppReposAction> => {
   return async (dispatch, getState) => {
     const {
-      clusters: {
-        currentCluster,
-        clusters: {
-          [currentCluster]: { currentNamespace },
-        },
-      },
+      clusters: { currentCluster },
     } = getState();
     try {
       await AppRepository.delete(currentCluster, namespace, name);
-      dispatch(fetchRepos(currentNamespace));
```

Comment:
it's not necessary to re-fetch repos here since the component is already doing this (also it's out of the scope for the function that deletes a repo)

Reply:
It's not clear to me how the component does this, since the callback you've created is dependent on `namespace` and `kubeappsNamespace` but not any current state.

I realise I must be wrong (as you've tested this) but afaict, the `useEffect` you've created will only run your callback if the dependencies to that callback change, I thought?

Edit: OK, I think I see it's because the callback `refetchRepos` is explicitly called when the component does a delete. It's still unclear to me why the callback still executes (rather than returning the memoized result) given that the dependencies it lists haven't changed?

Reply:
> Edit: OK, I think I see it's because the callback refetchRepos is explicitly called when the component does a delete. It's still unclear to me why the callback still executes (rather than returning the memoized result) given that the dependencies it lists haven't changed?

Yes, that's a bit tricky, what's memoized is the "body" of the function, not the function itself. I needed to do this because if not, the reference to `refetchRepos` would be different with each render. With this trick, the `useEffect` hook is only executed once. When, for example, the `namespace` changes, the `refetchRepos` ref changes and it gets executed again.

Reply:
Great, thanks - the comment makes it much clearer.